rotkehlchen/tests/unit/test_trades.py

```diff
@@ -137,14 +138,61 @@ def test_query_trade_history_online_but_exchange_excluded(events_historian):
     that exchange is not connected, but is also at the excluded exchanges we
     don't end up querying trades of all exchanges"""
 
-    patch_latest_trades = patch.object(events_historian, 'query_location_latest_trades')
-    patch_iterate_exchanges = patch.object(events_historian.exchange_manager, 'iterate_exchanges')
+    exchange = create_test_binance(
+        name='Binance',
+        database=events_historian.db,
+        msg_aggregator=events_historian.msg_aggregator,
+    )
 
-    with patch_latest_trades as latest_trades_mock, patch_iterate_exchanges as iterate_exchanges_mock:  # noqa: E501
+    with (
+        patch.object(
+            target=events_historian.exchange_manager,
+            attribute='iterate_exchanges',
+        ) as iterate_exchanges_mock,
+        patch.object(
+            target=events_historian.exchange_manager,
+            attribute='connected_exchanges',
+            return_value={Location.BINANCE: [exchange]},
+        ),
+        patch.object(
+            target=exchange,
+            attribute='query_trade_history',
+        ) as patch_query_trade_history,
+    ):
         events_historian.query_trades(
             filter_query=TradesFilterQuery.make(location=Location.BINANCE),
             only_cache=False,
         )
 
-    assert latest_trades_mock.call_count == 0
+    assert patch_query_trade_history.call_count == 0
     assert iterate_exchanges_mock.call_count == 0
+
+
+@pytest.mark.parametrize('db_settings', [
+    {'non_syncing_exchanges': [ExchangeLocationID(name='Coinbase', location=Location.COINBASE)]}])
+def test_query_trade_history_with_exchange_instance_excluded(events_historian):
+    """Test that when ignoring the locations along with names, only the ignored location with that
+    name should be ignored and not all the instances of that location."""
```

Comment:
update docstring. Notice that I create a tuple insteaad of a list for the list comprehension. Here probably is not noticeable but tuples are faster than lists

```python
@pytest.mark.parametrize('db_settings', [
    {'non_syncing_exchanges': [ExchangeLocationID(name='Coinbase', location=Location.COINBASE)]}])
def test_query_trade_history_with_exchange_instance_excluded(events_historian):
    """
    Test that when an exchange is ignored and more instances of the same location exist
    we only ignore the correct instance and not all.
    """
    coinbase_instances = [create_test_coinbase(
        name=name,
        database=events_historian,
        msg_aggregator=events_historian.msg_aggregator,
    ) for name in ('Coinbase', 'Coinbase2')]

    events_historian.exchange_manager.connected_exchanges[Location.COINBASE] = coinbase_instances
    with (
        patch.object(target=coinbase_instances[0], attribute='query_trade_history') as coinbase_mock,  # noqa: E501
        patch.object(target=coinbase_instances[1], attribute='query_trade_history') as coinbase2_mock,  # noqa: E501
    ):
        events_historian.query_trades(
            filter_query=TradesFilterQuery.make(location=Location.COINBASE),
            only_cache=False,
        )

    assert coinbase_mock.call_count == 0
    assert coinbase2_mock.call_count == 1
```